subsys/net/lib/nrf_cloud/src/nrf_cloud_fota.c

```diff
@@ -0,0 +1,928 @@
+			break;
+		}
+
+		/* Reset if new progress is less than previous */
+		if (current_fota.sent_dl_progress >
+		    current_fota.dl_progress) {
+			current_fota.sent_dl_progress = 0;
+		}
+
+		/* Only send progress update when finished and if increment
+		 * threshold is met
+		 */
+		if (current_fota.dl_progress != 100 &&
+		    ((current_fota.dl_progress -
+		      current_fota.sent_dl_progress) <
+		     CONFIG_NRF_CLOUD_FOTA_PROGRESS_PCT_INCREMENT)) {
+			break;
+		}
+
+		current_fota.sent_dl_progress = current_fota.dl_progress;
+		send_job_update(&current_fota);
+		break;
+	default:
+		break;
+	}
+
+	last_fota_dl_evt = evt->id;
+}
+
+static int parse_job(struct nrf_cloud_fota_job * const job)
+{
+	char *save_ptr = NULL;
+	char *end_ptr = &job->mqtt_payload[job->mqtt_payload_size - 1];
+	char *tok;
+
+	/* Job format:
+	 * [“jobExecutionId”,firmwareType,fileSize,”host”,”path”]
+	 *
+	 * Example:
+	 * [“abcd1234”,0,1234,”nrfcloud.com”,"v1/firmwares/appfw.bin"]
+	 */
+	LOG_DBG("Parsing: %s", log_strdup(job->mqtt_payload));
+
+	/* Get execution ID */
+	job->id = strtok_r(job->mqtt_payload, "[\",]", &save_ptr);
+	if (!job->id) {
+		goto handle_error;
+	}
+
+	/* Get fw type */
+	tok = strtok_r(NULL, ",", &save_ptr);
+	if (!tok) {
+		goto handle_error;
+	}
+	errno = 0;
+	job->type = (int)strtol(tok, &end_ptr, 10);
+	if (errno) {
+		goto handle_error;
+	}
+	if (job->type < NRF_FOTA_TYPE__FIRST ||
+	    job->type >= NRF_FOTA_TYPE__INVALID) {
+		LOG_ERR("Invalid FOTA type: %d", job->type);
+		goto handle_error;
+	}
+
+	/* Get file size */
+	tok = strtok_r(NULL, ",", &save_ptr);
+	if (!tok) {
+		goto handle_error;
+	}
+	errno = 0;
+	job->file_size = (int)strtol(tok, &end_ptr, 10);
+	if (errno) {
+		goto handle_error;
+	}
+
+	/* Get hostname */
+	job->host = strtok_r(NULL, "\",", &save_ptr);
+	if (!job->host) {
+		goto handle_error;
+	}
+
+	/* Get file path
+	 * Note: MCUboot bootloader updates can have
+	 * two paths separated by a space
+	 */
+	job->path = strtok_r(NULL, "\",]", &save_ptr);
+	if (!job->path) {
+		goto handle_error;
+	}
+
+	return 0;
+
+handle_error:
+
+	nrf_cloud_free(job->mqtt_payload);
+	memset(job, 0, sizeof(*job));
+
+	return -EINVAL;
+}
+
+static void send_event(const enum nrf_cloud_fota_evt_id id,
+		       const struct nrf_cloud_fota_job * const job)
+{
+	__ASSERT_NO_MSG(job != NULL);
+
+	struct nrf_cloud_fota_evt evt = {
+		.id = id,
+		.status = job->status,
+		.type = job->type
+	};
+
+	switch (id) {
+	case NRF_FOTA_EVT_ERROR:
+		evt.evt_data.error = job->error;
+		break;
+	case NRF_FOTA_EVT_DL_PROGRESS:
+		evt.evt_data.dl_progress = job->dl_progress;
+		break;
+	default:
+		break;
+	}
+
+	if (event_cb) {
+		event_cb(&evt);
+	}
+
+}
+
+static int start_job(struct nrf_cloud_fota_job * const job)
+{
+	__ASSERT_NO_MSG(job != NULL);
+
+	int ret;
+	int sec_tag = -1;
+	int fragment_size = 0;
+
+#if IS_ENABLED(CONFIG_NRF_CLOUD_FOTA_HTTPS_DOWNLOADS)
+	sec_tag = CONFIG_NRF_CLOUD_SEC_TAG;
+	fragment_size = 1024;
```

Comment:
I think this can be increased, depending on the typical HTTP header size from AWS. Perhaps it should even be provided in the job description somehow to allow customers to fine-tune the fragment size to decrease downloaded bytes and increase speed

Reply:
yes i agree this can be increased (or made into a config option).  once we get fota v2 pushed to dev or beta, i will test with a larger size.